### PR TITLE
refactor(hub-common): remove deprecated hub-content package

### PR DESCRIPTION
**Pull request by tomwayson**
1. Description:

1. Instructions for testing:

1. Part of [3319](https://devtopia.esri.com/dc/hub/issues/3319)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

### CODE REVIEW

File: packages/common/src/content/_internal.ts [hunk 1]
```diff
@@ -183,6 +183,8 @@ const getSolutionUrl = (
       // deployed solutions go to the generic content route
       hubUrl = `/content/${identifier}`;
     }
+    // TODO: I think this is a bug. The line below should be w/in else
+    // in order to avoid overwriting the above for deployed solutions, right?
```
Comment (dbouwman) [inline packages/common/src/content/_internal.ts:187]: I think you are correct @tomwayson

File: packages/common/src/content/index.ts [hunk 2]
```diff
@@ -438,11 +438,12 @@ export const getContentTypeLabel = (
 };
 
 /**
- * returns a new content that has the specified hubId and updated identifier
+ * DEPRECATED returns a new content that has the specified hubId and updated identifier
```
Comment (dbouwman) [inline packages/common/src/content/index.ts:441]: If this PR removes a package... which I expect is breaking change... can we remove these functions as well?

**Review by dbouwman — APPROVED** [review]
This seems ok to me, but the stats on the PR seem weird... this only has 43 files changes, but it's saying 84,000 lines added? 
image
Before merging, double check that it's some O_O diff thing. Also if we're going to do a major with this, let's kill more deprecated fns

File: packages/common/src/content/index.ts (hunk 2 above)
Comment (tomwayson) [reply packages/common/src/content/index.ts:441]: it's not a breaking change to remove the package from source code - the package still exists on NPM and is unchanged. 😉

File: packages/common/src/content/_internal.ts (hunk 1 above)
Comment (tomwayson) [reply packages/common/src/content/_internal.ts:187]: thx. I wonder if we have some code in -ui that is correcting for this bug. Maybe @vivzhang knows?

File: packages/common/src/content/index.ts (hunk 2 above)
Comment (dbouwman) [reply packages/common/src/content/index.ts:441]: 🎉🎉🎉🎉

**Comment by tomwayson** [comment]
the large diff is due to package-lock.json:

image

I would have expected it to be red, I suspect the reason it's green is that by adding the demo packages back in, package-lock.json includes all their (dev) dependencies like webpack, Ember, etc, which as we know are quite hefty.

**Pull request merged**